### PR TITLE
Implement insertion matrix (and add to README)

### PR DESCRIPTION
**Pull request by prm-james-hill**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Python client SDK for the Promoted.ai Delivery API
 - Demonstrates and implements the recommended practices and data types for calling Promoted.ai's Delivery API.
 - Client-side position assignment and paging when not using results from Delivery API, for example when logging only or as part of an experiment control.
 
-## Creating a PromotedDelieryClient
+## Creating a PromotedDeliveryClient
```
Comment (prm-dan) [inline README.md:10]: Unrelated to this line, we'll also want to bump the versions of the SDK.

File: README.md [hunk 2]
```diff
@@ -92,6 +93,40 @@ Field Name | Type | Optional? | Description
...
+]
+req = Request(insertion_matrix_headers=insertion_matrix_headers,
+              insertion_matrix=insertion_matrix,
+              ...
+)
+```
+
+Things to note:
+* Properties with nested `struct`s should concatenate paths with a `.` (period).
+* Properties which don't exist for an insertion must be specified as `None`.
```
Comment (prm-dan) [inline README.md:126]: This is an interesting point.  We should chat about what this does server-side.  I would expect this to behavior similarly as having a prop of `"price": None`.

File: promoted_python_delivery_client/client/delivery_request_state.py [hunk 3]
```diff
@@ -14,6 +14,8 @@ def get_response_to_return(self, resp: Response) -> Response:
 
     def get_request_to_send(self, max_request_insertions: int) -> Request:
         req = self.delivery_request.request
-        if len(req.insertion) > max_request_insertions:
+        if req.insertion is not None and len(req.insertion) > max_request_insertions:
```
Comment (prm-dan) [inline promoted_python_delivery_client/client/delivery_request_state.py]: I think you can also do

```
if req.insertion and len(req.insertion) > max_request_insertions:
```

**Review by prm-dan — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (prm-james-hill) [reply README.md:10]: Sounds like this is done as part of the release process, so not doing anything here for it

File: promoted_python_delivery_client/client/delivery_request_state.py (hunk 3 above)
Comment (prm-james-hill) [reply promoted_python_delivery_client/client/delivery_request_state.py]: Done here and elsewhere

**Pull request merged**